omnigibson/tasks/grasp_task.py

```diff
@@ -0,0 +1,242 @@
+from scipy.spatial.transform import Rotation as R
+
+import omnigibson as og
+import omnigibson.utils.transform_utils as T
+from omnigibson.action_primitives.starter_semantic_action_primitives import (
+    PlanningContext,
+    StarterSemanticActionPrimitives,
+)
+from omnigibson.macros import gm
+from omnigibson.objects.object_base import REGISTERED_OBJECTS
+from omnigibson.reward_functions.grasp_reward import GraspReward
+from omnigibson.scenes.scene_base import Scene
+from omnigibson.tasks.task_base import BaseTask
+from omnigibson.termination_conditions.falling import Falling
+from omnigibson.termination_conditions.grasp_goal import GraspGoal
+from omnigibson.termination_conditions.timeout import Timeout
+from omnigibson.utils.grasping_planning_utils import get_grasp_poses_for_object_sticky
+from omnigibson.utils.motion_planning_utils import set_arm_and_detect_collision
+from omnigibson.utils.python_utils import classproperty, create_class_from_registry_and_config
+from omnigibson.utils.sim_utils import land_object
+
+MAX_JOINT_RANDOMIZATION_ATTEMPTS = 50
+
+
+class GraspTask(BaseTask):
+    """
+    Grasp task
+    """
+
+    def __init__(
+        self, obj_name, termination_config=None, reward_config=None, precached_reset_pose_path=None, objects_config=None
+    ):
+        self.obj_name = obj_name
+        self._primitive_controller = None
+        self._reset_poses = None
+        self._objects_config = objects_config
+        if precached_reset_pose_path is not None:
+            with open(precached_reset_pose_path) as f:
+                self._reset_poses = json.load(f)
+        super().__init__(termination_config=termination_config, reward_config=reward_config)
+
+    def _load(self, env):
+        for obj_config in self._objects_config:
+            obj = env.scene.object_registry("name", obj_config["name"])
+            # Create object
+            if obj is None:
+                obj = create_class_from_registry_and_config(
+                    cls_name=obj_config["type"],
+                    cls_registry=REGISTERED_OBJECTS,
+                    cfg=obj_config,
+                    cls_type_descriptor="object",
+                )
+                # Import the object into the simulator and set the pose
+                env.scene.add_object(obj)
+
+            obj_pos = [0.0, 0.0, 0.0] if "position" not in obj_config else obj_config["position"]
+            obj_orn = [0.0, 0.0, 0.0, 1.0] if "orientation" not in obj_config else obj_config["orientation"]
+            obj_pos, obj_orn = T.pose_transform(*env.scene.prim.get_position_orientation(), obj_pos, obj_orn)
+            obj.set_position_orientation(obj_pos, obj_orn)
+
+    def _create_termination_conditions(self):
+        terminations = dict()
+        # terminations["graspgoal"] = GraspGoal(
+        #     self.obj_name
+        # )
+        terminations["timeout"] = Timeout(max_steps=self._termination_config["max_steps"])
+        # terminations["falling"] = Falling()
```

Comment:
why are some of the conditions commented out?

Reply:
having resets happen at different times is very tricky. this prevents that from happening. @hang-yin please leave a note here.